### ISSUE TITLE
Getting response as 'invalid_client'

### ISSUE THREAD

**Issue by rahimnagori** [XeroAPI/xero-php-oauth2#10]
I'm able to successfully generate the authorization URL.
App is being authorized.
Redirected back to my application.
But at the step : 
```
$accessToken = $provider->getAccessToken('authorization_code', [
        'code' => $_GET['code']
      ]);
```
I'm getting the response / error as  : 'invalid_client'.

Any idea what I'm doing wrong. I have crossed checked the credentials, that are correct.

**Comment by SidneyAllen** [comment]
Hi @rahimnagori 

According to our docs.

If the client_id or client_secret you pass in are incorrect then you will receive this response.

{
    "error": "invalid_client"
}
To solve this error, make sure you have the correct credentials for your app. Double check your client id in My Apps and try generating a new secret if required.

**Comment by rahimnagori** [comment]
Thank you for the reply. I have checked my credentials and it was correct the problem was just a silly one similar to this I was able to resolve.
Executing composer require xeroapi/xero-php-oauth2 shows the following error now : 

- Installation request for xeroapi/xero-php-oauth2 ^1.0 -> satisfiable by xeroapi/xero-php-oauth2[1.0.0].
- Conclusion: remove league/oauth2-client 1.4.2
- Conclusion: don't install league/oauth2-client 1.4.2
- xeroapi/xero-php-oauth2 1.0.0 requires league/oauth2-client ^2.4 -> satisfiable by league/oauth2-client[2.4.0, 2.4.1].
- Can only install one of: league/oauth2-client[2.4.0, 1.4.2].
- Can only install one of: league/oauth2-client[2.4.1, 1.4.2].
- Installation request for league/oauth2-client (locked at 1.4.2) -> satisfiable by league/oauth2-client[1.4.2].

Could you please help me with this?

**Comment by SidneyAllen** [comment]
Hi @rahimnagori 

I've updated xero-php-oauth2 package to use league/oauth2-client (2.4.1).  I went to an empty dir, ran composer init, then ran composer require xeroapi/xero-php-oauth2

and got the below in my terminal

LM-CXXXXXQT:PHP-test sid.maestre$ composer require xeroapi/xero-php-oauth2
Using version ^1.0 for xeroapi/xero-php-oauth2
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Installing paragonie/random_compat (v9.99.99)
    Downloading: 100%         

  - Installing ralouphie/getallheaders (3.0.3)
    Loading from cache

  - Installing psr/http-message (1.0.1)
    Downloading: 100%         

  - Installing guzzlehttp/psr7 (1.6.1)
    Loading from cache

  - Installing guzzlehttp/promises (v1.3.1)
    Downloading: 100%         

  - Installing guzzlehttp/guzzle (6.4.1)
    Loading from cache

  - Installing league/oauth2-client (2.4.1)
    Downloading: 100%         

  - Installing xeroapi/xero-php-oauth2 (1.0.1)
    Downloading: 100%         

paragonie/random_compat suggests installing ext-libsodium (Provides a modern crypto API that can be used to generate random bytes.)
guzzlehttp/psr7 suggests installing zendframework/zend-httphandlerrunner (Emit PSR-7 responses)
guzzlehttp/guzzle suggests installing psr/log (Required for using the Log middleware)
Writing lock file
Generating autoload files

I'm not a composer expert.  Sounds like you had league/oauth2-client 1.4.2 already in your composer project?
